doc/user/content/sql/explain-timestamp.md

```diff
@@ -0,0 +1,192 @@
+
+### Output format
+
+You can select between `JSON` and `TEXT` for the output format of `EXPLAIN TIMESTAMP`. Non-text
+output is more machine-readable and can be parsed by common graph visualization libraries,
+while formatted text is more human-readable.
+
+Output type | Description
+------|-----
+**TEXT** | Format the explanation output as UTF-8 text.
+**JSON** | Format the explanation output as a JSON object.
+
+## Details
+
+The explanation is divided in two parts:
+
+1. Determinations for a timestamp
+2. Sources frontiers
+
+Having a _query timestamp_ outside the _[read, write)_ frontier values of a source can explain the presence of delays. While in the middle, the space of processed but not yet compacted data, allows building and returning a correct result immediately.
+
+### Determinations for a timestamp
+
+Queries in Materialize have a logical timestamp, known as _query timestamp_. It plays a critical role to return a correct result. Returning a correct result implies retrieving data with the same logical time from each source present in a query.
+
+In this case, sources are objects providing data: materialized views, views, indexes, tables, or sources. Each will have a pair of logical timestamps frontiers, denoted as _sources frontiers_.
+
+This section contains the following fields:
+
+Field | Meaning | Example
+---------|---------|---------
+**query timestamp** | The query timestamp value |`1673612424151 (2023-01-13 12:20:24.151)`
+**oracle read** | The value of the timeline's oracle timestamp, if used. | `1673612424151 (2023-01-13 12:20:24.151)`
+**largest not in advance of upper** | The largest timestamp not in advance of upper. | `1673612424151 (2023-01-13 12:20:24.151)`
+**since** | The maximum read frontier of all involved sources. | `[1673612423000 (2023-01-13 12:20:23.000)]`
+**upper** | The minimum write frontier of all involved sources | `[1673612424152 (2023-01-13 12:20:24.152)]`
+**can respond immediately** | Returns true when the **query timestamp** is greater or equal to **since** and lower than **upper** | `true`
+**timeline** | The type of timeline the query's timestamp belongs | `Some(EpochMilliseconds)`
+
+Possible timeline values are:
+
+Field | Meaning | Example
+---------|---------|---------
+**EpochMilliseconds** | Means the timestamp is the number of milliseconds since the Unix epoch. | `Some(EpochMilliseconds)`
+**External** | Means the timestamp comes from an external data source and we don't know what the number means. The attached String is the source's name, which will result in different sources being incomparable. | `Some(External)`
+**User** | Means the user has manually specified a timeline. The attached String is specified by the user, allowing them to decide sources that are joinable. | `Some(User)`
+
+### Sources frontiers
+
+Every source has a beginning _read frontier_ and an ending _write frontier_. They stand for a source’s limits to return a correct result immediately:
```

Comment:
It's possible for upper < since. Not sure we should change this text to talk about that though.